src/utils/helpers/focus-trap/focus-trap.js

```diff
@@ -0,0 +1,49 @@
+const blockTabbing = (ev) => {
+  if (ev.keyCode === 9 || ev.keyCode === 16) {
+    ev.preventDefault();
+  }
+};
+
+const tabbing = (ev, firstFocusableElement, lastFocusableElement) => {
+  if (ev.key === 'Tab' || ev.keyCode === 9) {
+    if (ev.shiftKey) /* shift + tab */ {
+      if (document.activeElement === firstFocusableElement) {
+        lastFocusableElement.focus();
+        ev.preventDefault();
+      }
+    } else if (document.activeElement === lastFocusableElement) {
+      firstFocusableElement.focus();
+      ev.preventDefault();
+    }
+  }
+};
+
+const setFocusTrap = (element) => {
+  // eslint-disable-next-line max-len
+  const ALL_FOCUSABLE_ELEMENTS = 'button, [href], input:not([type="hidden"]), select, textarea, [tabindex]:not([tabindex="-1"])';
+  const focusableElements = element.querySelectorAll(ALL_FOCUSABLE_ELEMENTS);
+
+  if (focusableElements.length <= 0) {
+    document.addEventListener('keydown', blockTabbing);
+    return;
+  }
+
+  const firstFocusableElement = focusableElements[0];
+  const lastFocusableElement = focusableElements[focusableElements.length - 1];
+
+  setTimeout(() => {
```

Comment:
Why is this in setTimout?

Reply:
without setTimeout focus won't land on the first focusable item when the dialog was opening. this async resolved the problem

Reply:
This relies upon react's batched updates being applied before the next "tick". React now has low/high priority scheduling using rAF. I think this is prone to errors and another approach should be used. Im happy to be corrected though.

Instead, you should add a lifecycle method to the dialog.

Side note,
W3C recommends listening to the focus change on the document and restoring the focus instead. Did you try that approach?

https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/js/dialog.js

This is reactive to the DOM update instead of trying to prevent the focus change.

Reply:
setTimouet has been removed